application/frontend/controllers/MfaController.php

```diff
@@ -57,13 +57,19 @@ public function actionCreate()
     /**
      * Verify value with MFA backend
      * @param int $id
+     * @param string $type (optional) If not blank, it must be `registration`, referring to verifying a webauthn registration
      * @throws BadRequestHttpException
      * @throws NotFoundHttpException
      * @throws TooManyRequestsHttpException
      * @return null
      */
-    public function actionVerify(int $id)
+    public function actionVerify(int $id, string $type = "")
     {
+
+        if ($type != "" && $type != Mfa::VERIFY_REGISTRATION) {
+            throw new BadRequestHttpException('Verify type can only be: ' . Mfa::VERIFY_REGISTRATION . " Got: $type." , 1660222364);
+        }
+
```

Comment:
I think this check could probably be left to the Mfa->verify() or to the MfaBackend(Type)->verify() function, rather than putting it up here in the controller. Just a suggestion.